ballerina-tests/tests/04_input_parameters.bal

```diff
@@ -17,105 +17,31 @@
 import ballerina/test;
 import ballerina/lang.value;
 
-const float CONVERSION_KG_TO_LBS = 2.205;
-
-@test:Config {
-    groups: ["inputs"]
-}
-isolated function testFunctionsWithInputParameter() returns error? {
-    string document = string`{ greet (name: "Thisaru") }`;
-    string url = "http://localhost:9091/inputs";
-    json actualPayload = check getJsonPayloadFromService(url, document);
-    json expectedPayload = {
-        data: {
-            greet: "Hello, Thisaru"
-        }
-    };
-    assertJsonValuesWithOrder(actualPayload, expectedPayload);
-}
-
-@test:Config {
-    groups: ["inputs"]
-}
-isolated function testInputParameterTypeNotPresentInReturnTypes() returns error? {
-    string document = "{ isLegal(age: 21) }";
-    string url = "http://localhost:9091/inputs";
-    json actualPayload = check getJsonPayloadFromService(url, document);
-    json expectedPayload = {
-        data: {
-            isLegal: true
-        }
-    };
-    assertJsonValuesWithOrder(actualPayload, expectedPayload);
-}
-
-@test:Config {
-    groups: ["inputs", "validation"]
-}
-isolated function testInvalidArgument() returns error? {
-    string document = "{ quote(id: 4) }";
-    string url = "http://localhost:9091/inputs";
-    json actualPayload = check getJsonPayloadFromBadRequest(url, document);
-    json expectedPayload = {
-        errors: [
-            {
-                message: string`Unknown argument "id" on field "Query.quote".`,
-                locations: [
-                    {
-                        line: 1,
-                        column: 9
-                    }
-                ]
-            }
-        ]
-    };
-    assertJsonValuesWithOrder(actualPayload, expectedPayload);
-}
-
-@test:Config {
-    groups: ["inputs"]
-}
-isolated function testQueryWithoutDefaultParameter() returns error? {
-    string document = "{ quoteById }";
-    string url = "http://localhost:9091/inputs";
-    json actualPayload = check getJsonPayloadFromService(url, document);
-    json expectedPayload = {
-        data: {
-            quoteById: "I am a high-functioning sociapath!"
-        }
-    };
-    assertJsonValuesWithOrder(actualPayload, expectedPayload);
-}
-
 @test:Config {
-    groups: ["inputs"]
+    groups: ["inputs"],
+    dataProvider: dataProviderInputParameters
 }
-isolated function testQueryWithDefaultParameter() returns error? {
-    string document = "{ quoteById(id: 2) }";
+isolated function testInputParameters(string documentFileName) returns error? {
```

Comment:
A better name would be `resourceFileName` since we use this as the document and the expected result. Check other places as well.